ark/utils/metacluster_remap_gui/metaclustergui.py

```diff
@@ -0,0 +1,463 @@
+            if cluster in self.selected_clusters:
+                return 1
+            else:
+                return 0
+        return [[is_selected(c) for c in self.mcd.clusters.index]]
+
+    def update_gui(self):
+        self.im_cs.set_data(self.selection_mask)
+        self.im_cs.set_extent((0, self.mcd.cluster_count, 0, 1))
+
+        if not self._heatmaps_stale:
+            print("skipping other repaints")
+            self.fig.canvas.draw()
+            return
+
+        self.normalizer.calibrate(self.preplot(self.mcd.clusters).values)
+
+        # clusters heatmap
+        self.im_c.set_data(self.preplot(self.mcd.clusters))
+        self.im_c.set_extent((0, self.mcd.cluster_count, 0, self.mcd.marker_count))
+        self.im_c.set_clim(self.normalizer.vmin, self.normalizer.vmax)
+
+        # metaclusters heatmap
+        self.im_m.set_data(self.preplot(self.mcd.metaclusters))
+        self.im_m.set_extent((0, self.mcd.metacluster_count, 0, self.mcd.marker_count))
+        self.im_m.set_clim(self.normalizer.vmin, self.normalizer.vmax)
+
+        # xaxis metacluster color labels
+        assert max(self.mcd.metaclusters.index) <= self.mcd.cluster_count, \
+            "Can't support metaclusters idx > cluster count"
+        self.im_cl.set_data([self.mcd.clusters_with_metaclusters['metacluster']])
+        self.im_cl.set_extent((0, self.mcd.cluster_count, 0, 1))
+        self.im_cl.set_cmap(self.cmap)
+        self.ax_ml.set_xticks(np.arange(self.mcd.metacluster_count)+0.5)
+        self.ax_ml.set_xticklabels(self.mcd.metacluster_displaynames, rotation=90, fontsize=7)
+        self.im_ml.set_data([self.mcd.metaclusters.index])
+        self.im_ml.set_extent((0, self.mcd.metacluster_count, 0, 1))
+        self.im_ml.set_cmap(self.cmap)
+
+        # xaxis pixelcount graphs
+        ax_cp_ymax = max(self.mcd.cluster_pixelcounts['count'])*1.65
+        self.ax_cp.set_ylim(0, ax_cp_ymax)
+        sorted_pixel_counts = self.mcd.clusters.join(self.mcd.cluster_pixelcounts)['count']
+        for rect, h in zip(self.rects_cp, sorted_pixel_counts):
+            rect.set_height(h)
+        for label, y in zip(self.labels_cp, sorted_pixel_counts):
+            text = "{:0.0f}".format(y / 1000)
+            label_y_spacing = ax_cp_ymax * 0.05
+            label.set_y(y + label_y_spacing)
+            label.set_text(text)
+
+        self.fig.canvas.draw()
+        self._heatmaps_stale = False
+
+    def enable_debug_mode(self):
+        self.fig.canvas.footer_visible = True
+        DEBUG_VIEW.clear_output()
+        DEBUG_VIEW.append_stdout("Debug mode started\n")
+        display(DEBUG_VIEW)
+
+    def remap_current_selection(self, metacluster):
+        for cluster in self.selected_clusters:
+            print('remapping', cluster, metacluster)
+            self.mcd.remap(cluster, metacluster)
+        self._heatmaps_stale = True
+        self.mcd.save_output_mapping()
+
+    @DEBUG_VIEW.capture(clear_output=False)
```

Comment:
Do you think the `DEBUG_VIEWs` here are useful to keep in case we need to make changes and debug? Or should they be removed

Reply:
They are disabled by default unless turned on. I did build the debug view as a first class feature meant to left in. Toggling is documented in the developer notes here:
![image](https://user-images.githubusercontent.com/6101677/134822116-69189ece-133f-4cd2-9eff-4fece9d59b84.png)